CHANGELOG.md

```diff
@@ -2,6 +2,8 @@
 
 ## 0.6.0 (TBD)
 
+- Adds the `TransactionProver` trait (#865).
+- [BREAKING] Renames the `TransactionProver` struct to `LocalTransactionProver` (#865).
```

Comment:
nit: I think the convention is to write these in past tense. Additionally I'd merge both into a single changelog line but I it's likely not a problem if they are kept separated.